tests/admin_scripts/tests.py

```diff
@@ -236,6 +236,13 @@ def test_builtin_with_bad_environment(self):
         self.assertNoOutput(out)
         self.assertOutput(err, "No module named '?bad_settings'?", regex=True)
 
+    def test_default_module_settings_environment(self):
+        "invalid env var: django-admin builtin commands succeed if environment variable is wrongly set"
+        with mock.patch.dict('os.environ', {'DJANGO_SETTINGS_MODULE': 'wrong_path'}):
+            args = ['startproject']
+            out, err = self.run_django_admin(args, settings_file=os.environ['DJANGO_SETTINGS_MODULE'])
+            self.assertOutput(err, "You must provide a project name", regex=True)
+
```

Comment:
so it turns out using `mock.patch.dict` isn't necessary, because `run_django_admin` is actually setting up `os.environ` for the sub-command. so you can refactor this test to just:

```py
out, err = self.run_django_admin(['startproject'], 'bad_settings')
self.assertOutput(err, "You must provide a project name", regex=True)
```

Reply:
( see https://github.com/django/django/blob/master/tests/admin_scripts/tests.py#L141 )